app/models/miq_provision_virt_workflow.rb

```diff
@@ -395,26 +395,26 @@ def allowed_cat_entries(options)
   end
 
   def allowed_vlans(options = {})
-    @allowed_vlan_cache ||= available_vlans(options)
+    @allowed_vlan_cache ||= available_vlans_and_hosts(options)[0]
     filter_by_tags(@allowed_vlan_cache, options)
   end
 
-  def available_vlans(options = {})
+  def available_vlans_and_hosts(options = {})
```

Comment:
does this really need to return a tuple of `vlans, hosts` ?
I dont see the return value of `hosts` being used anywhere.
My guess is, that you will end up with cleaner code, if you try to introduce a second method `available_hosts`